### PR TITLE
Add fleet-server package

### PR DESCRIPTION
**Pull request by ruflin**
NOTE: This is not working at all yet. No review please. This is used to collaborate on the changes.

This package contains all the assets needed by fleet-server.

### CODE REVIEW

File: packages/fleet_server/elasticsearch/index_template/fleet_policy.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (ruflin) [inline packages/fleet_server/elasticsearch/index_template/fleet_policy.json:1]: Here is an example on how this should look: https://github.com/elastic/package-storage/blob/production/packages/endpoint/0.17.0/elasticsearch/index_template/metrics-metadata-current.json

I wonder what happens if we prefix this template with a `.`. Not even sure it is possible.

**Comment by ruflin** [comment]
Indication on the templates we need:
```
{"level":"info","name":".fleet-enrollment-api-keys","time":"Jan 19 15:21:37.411723","message":"Create template"}
{"level":"info","name":".fleet-policies","time":"Jan 19 15:21:37.648501","message":"Create template"}
{"level":"info","name":".fleet-policies-leader","time":"Jan 19 15:21:37.784996","message":"Create template"}
{"level":"info","name":".fleet-servers","time":"Jan 19 15:21:37.921092","message":"Create template"}
{"level":"info","name":".fleet-actions","time":"Jan 19 15:21:38.094132","message":"Create template"}
{"level":"info","policy":".fleet-actions-results-ilm-policy","time":"Jan 19 15:21:38.239051","message":"ILM policy is not found. Create a new one."}
{"level":"info","name":".fleet-actions-results","time":"Jan 19 15:21:38.286755","message":"Create template"}
{"level":"info","name":".fleet-agents","time":"Jan 19 15:21:38.588401","message":"Create template"}
```

One of the questions will be if we can make the templates for it private.

@aleksmaus Which of the above are indices and which are data streams?

**Comment by aleksmaus** [comment]
@ruflin we only have one datastream for now .fleet-actions-results
and btw we can't use this on elasticsearch 7.10 since the datastream are not allowed to start with ".", only 7.11

**Comment by ruflin** [comment]
@aleksmaus This package will only be compatible with ^7.12 so should not be an issue. I suggest for the first iteration we put all templates globally and don't use the `data_stream` directories? This does not mean we can't use data-streams.

**Comment by aleksmaus** [comment]
Closing this draft. Replaced with https://github.com/elastic/integrations/pull/544.

**Pull request closed without merging**